src/app/user/user-details/user-details.component.html

```diff
@@ -0,0 +1,60 @@
+<div class="main-container">
+  @if (user$ | async; as user) {
+    <form class="details-container" [formGroup]="roleForm" (ngSubmit)="onSave(user)">
+      <div class="main-title-container">
+        <h1>User Details</h1>
+        @if (this.editable === false) {
+          <button mat-icon-button matTooltip="Edit Roles" type="button" (click)="onEdit(user.roles)">
+            <mat-icon>edit</mat-icon>
+          </button>
+        } @else {
+          <div class="controls-container">
+            <button mat-icon-button matTooltip="Cancel" type="button" (click)="onCancel()">
+              <mat-icon>close</mat-icon>
+            </button>
+            <button mat-icon-button matTooltip="Save" type="submit">
+              <mat-icon>check</mat-icon>
+            </button>
+          </div>
+        }
+      </div>
+      <div class="details-card">
+        <div class="details-item">
+          <div class="title-container">
+            <mat-icon>title</mat-icon>
+            <h2>First Name</h2>
+          </div>
+          <p>{{ user.firstName }}</p>
+        </div>
+        <div class="details-item">
+          <div class="title-container">
+            <mat-icon>title</mat-icon>
+            <h2>Last Name</h2>
+          </div>
+          <p>{{ user.lastName }}</p>
+        </div>
+        <div class="details-item">
+          <div class="title-container">
+            <mat-icon>badge</mat-icon>
+            <h2>Username</h2>
+          </div>
+          <p>{{ user.username }}</p>
+        </div>
+        <div class="details-item">
+          <div class="title-container">
+            <mat-icon>lock</mat-icon>
+            <h2>Roles</h2>
+          </div>
+          @if (this.editable === false) {
+            @for (role of user.roles; track $index) {
+              <p>{{ role }}</p>
+            }
+          } @else {
+            <p><mat-checkbox formControlName="user">User</mat-checkbox></p>
```

Comment:
maybe it's worth replacing checkboxes with multi select dropdown? then you can just pass the value of the dropdown with the selected roles instead of checking each form control individually and pushing it to the array